### PR TITLE
Moving Nexus + adding post step for seeding

### PR DESCRIPTION
**Pull request by oybed**
**NOTE:** see important note below re: merging. 

This PR moves the `configure-nexus` role from the deprecated repo `ansible-stacks`. 

The secondary purpose for the move is to make it more readily available for use by the `openshift-applier` and the upcoming pre/post steps. This will allow for a better seeding of the nexus content by using a post-step to call `configure-nexus`. 

*resolved* : This PR is marked as **WIP** as it has a dependency on the CASL PR for pre/post steps get merged:
https://github.com/redhat-cop/casl-ansible/pull/155

**NOTE:** When merging this PR, please ensure to use `Create a merge commit` to preserve the commit history brought over from `ansible-stacks`. 

@JoshuaTPritchett FYI

### CODE REVIEW

**Comment by sherl0cks** [comment]
@oybed ack - waiting on the casl pr...

**Comment by sherl0cks** [comment]
@oybed interesting side effect with the new role and the `galaxy_requirements` entry in the inventory. 
`TASK [make-projects-unique : Regexp replace on params files] *******************
fatal: [localhost]: FAILED! => {"failed": true, "msg": "'dict object' has no attribute 'content'"}`

can you please:
- rebase this PR with master
- add a check like the below to https://github.com/rht-labs/labs-ci-cd/blob/master/roles/make-projects-unique/tasks/main.yml#L6
```
  when:
  - item.content is defined
  - item.content != ""
```

**Comment by oybed** [comment]
@sherl0cks done. PTAL. 

Note: we'll have to make the changes in CASL for the `make-applier-projects-unique` role.

**Comment by sherl0cks** [comment]
@oybed fails with:

```
TASK [casl-ansible/roles/openshift-applier : Run ansible-galaxy to pull in dependency roles] ***

failed: [localhost -> localhost] (item=/tmp/workspace/labs-ci-cd-ci-for-labs-ci-cd-pipeline/labs-ci-cd/inventory/../galaxy_requirements.yml) => {"changed": true, "cmd": ["ansible-galaxy", "install", "-r", "/tmp/workspace/labs-ci-cd-ci-for-labs-ci-cd-pipeline/labs-ci-cd/inventory/../galaxy_requirements.yml", "-p", "/tmp/ansible.6B3fPT/"], 
"delta": "0:00:01.022470", 
"end": "2018-02-16 14:31:54.941492", "failed": true,
 "item": "/tmp/workspace/labs-ci-cd-ci-for-labs-ci-cd-pipeline/labs-ci-cd/inventory/../galaxy_requirements.yml", 
"msg": "non-zero return code", 
"rc": 1, 
"start": "2018-02-16 14:31:53.919022", 
"stderr": " [WARNING]: - labs-ci-cd was NOT installed successfully: this role does not\nappear to have a meta/main.yml file.\nERROR! - you can use --ignore-errors to skip failed roles and finish processing the list.", 
"stderr_lines": [" [WARNING]: - labs-ci-cd was NOT installed successfully: this role does not", "appear to have a meta/main.yml file.", "ERROR! - you can use --ignore-errors to skip failed roles and finish processing the list."],
 "stdout": 
"", "stdout_lines": []}
```

File: galaxy_requirements.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+
+# This is the Ansible Galaxy requirements file to pull in the correct
+# dependencies to support this inventory
+
+# From 'rht-labs/labs-ci-cd'
+- src: https://github.com/rht-labs/labs-ci-cd
```
Comment (sherl0cks) [inline galaxy_requirements.yml]: why is this needed? feels like a level of unnecessary recursion?

**Comment by oybed** [comment]
@sherl0cks it's failing because the repo doesn't have the `meta/main.yml` yet - it's a bit catch-22. So either temporarily update the `galaxy_requirements.yml` to point to my PR branch, or we can submit a small PR up front to get `meta/main.yml` added. It's related to your other comment, which I'll address separately.

**Comment by sherl0cks** [comment]
assuming for a minute we need to clone the repo a second time, we need to update the regex processing to hit that file then during the PR build. Currently it only processes the `params` directory. We'll get CI failures here in the event that changes are made to the nexus role (or any others that come after it)

File: inventory/group_vars/all.yml [hunk 2]
```diff
@@ -82,6 +83,11 @@ openshift_cluster_content:
     namespace: labs-ci-cd
     tags:
     - nexus
+    post_step:
```
Comment (sherl0cks) [inline inventory/group_vars/all.yml]: this should be `post_steps` according to: https://github.com/redhat-cop/casl-ansible/blob/master/roles/openshift-applier/tasks/process-content.yml#L31

File: roles/configure-nexus/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  - nexus_url is not defined or nexus_url|trim == ''
+
+- name: "Set nexus_url Result"
+  set_fact:
+    nexus_url: "{{ nexus_url_result.stdout }}"
+  when:
+  - nexus_url is not defined or nexus_url|trim == ''
+
+- name: "Ensure Nexus is available before attempting to communicate"
+  import_tasks: wait-for-nexus.yml
```
Comment (sherl0cks) [inline roles/configure-nexus/tasks/main.yml]: Did `wait-for-nexus.yml` get lost in the merge? It's missing and the role crashes

File: inventory/group_vars/all.yml (hunk 2 above)
Comment (oybed) [reply inventory/group_vars/all.yml]: yes, that needs to be `post_steps` - will fix (I believe that was a change post PR with multiple things happening in parallel)

File: roles/configure-nexus/tasks/main.yml (hunk 3 above)
Comment (oybed) [reply roles/configure-nexus/tasks/main.yml]: fixed

Comment (oybed) [reply roles/configure-nexus/tasks/main.yml]: it was due to the `.gitignore` changing a long the way (since this PR has been in progress for awhile)

Comment (sherl0cks) [inline roles/configure-nexus/tasks/main.yml]: `import_tasks` needs to be moved to `include`

File: roles/configure-nexus/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+- name: "Configured Repos"
+  include_tasks: configure-repos.yml
```
Comment (sherl0cks) [inline roles/configure-nexus/tasks/main.yml]: `include_tasks` probably should also move to `include`

File: inventory/group_vars/all.yml [hunk 5]
```diff
@@ -82,6 +83,11 @@ openshift_cluster_content:
     namespace: labs-ci-cd
     tags:
     - nexus
+    post_steps:
+    - role: "labs-ci-cd/roles/configure-nexus"
+      vars:
+        nexus_name: "nexus"
+        nexus_project: "labs-ci-cd"
```
Comment (sherl0cks) [inline inventory/group_vars/all.yml]: I'd like to remove the vars - I do not think they are needed in this case, we can use defaults and see below about project name

File: roles/configure-nexus/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,22 @@
+---
+
+- name: "Look up Nexus URL"
+  command: >
+    oc get route {{ nexus_name | default('nexus') }} -n {{ nexus_project }} -o custom-columns=ROUTE:.spec.host --no-headers
```
Comment (sherl0cks) [inline roles/configure-nexus/tasks/main.yml]: `nexus_project` as a variable here is problematic when we need to make project names unique. I tested that `content.namespace` is set and usable here. It's a bit of a leaky abstraction, but it will make life much easier if used here. I dont think this role needs to be portable, and if it does, then we can have logic to set the variable.
